src/core/Network.ts

```diff
@@ -35,26 +35,26 @@ export abstract class Network {
     public createAddressFromPublicKey(publicKey: Key): RawAddress {
         const publicKeyBytes = publicKey.toBytes();
         // step 1: sha3 hash of the public key
-        const publicKeyHash = this.addressHasher().update(Buffer.from(publicKeyBytes));
+        const publicKeyHash = this.addressHasher().arrayBuffer(publicKeyBytes);
 
         // step 2: ripemd160 hash of (1)
-        const ripemd160Hash = new ripemd160().update(publicKeyHash.digest()).digest();
+        const ripemd160Hash = new ripemd160().update(new Uint8Array(publicKeyBytes)).digest();
```

Comment:
Why the new Uint8Array(..), can be used directly... 
```
const ripemd160Hash = new ripemd160().update(publicKeyBytes).digest();
```

Reply:
I have rolledback to buffer.from, the ripemd160 update does not take arraybuffer directly